File: mobile/constants/theme.js
<<<<<<< HEAD:mobile/constants/theme.ts
// =======================
// COLORS
// =======================
import { PlatformColor } from 'react-native';

=======
>>>>>>> main:mobile/constants/theme.js
export const colors = {
  // Background colors
  background: '#0b132b',
  backgroundSecondary: '#0d1630',
  backgroundCard: '#1c2541',
  backgroundCardDark: '#151e38',
  backgroundOverlay: 'rgba(11,19,43,0.95)',
  
  // Gradient colors
  gradientStart: '#6c63ff',
  gradientEnd: '#8b7fff',
  gradientBlueStart: '#38c4fc',
  gradientBlueEnd: '#76caf5',
  
  // Text colors
  textPrimary: '#ffffff',
  textSecondary: '#a9b4d1',
  textAccent: '#cbb8ff',
  textBlue: '#68d4ff',
  textError: '#ff3b30',
  
  // UI colors
  border: 'rgba(255,255,255,0.1)',
  borderLight: 'rgba(255,255,255,0.05)',
  borderMedium: 'rgba(255,255,255,0.2)',
  
  // Accent colors
  accentPurple: 'rgba(108,99,255,0.1)',
  accentPurpleMedium: 'rgba(108,99,255,0.2)',
  accentBlue: 'rgba(104,212,255,0.1)',
  accentBlueMedium: 'rgba(104,212,255,0.2)',
  accentPurpleText: 'rgba(203,184,255,0.1)',
  accentError: 'rgba(255,59,48,0.1)',
  accentErrorMedium: 'rgba(255,59,48,0.2)',
  accentErrorBorder: 'rgba(255,59,48,0.5)',
  
  // Glass effect
  glassBackground: 'rgba(28,37,65,0.6)',
  glassBackgroundLight: 'rgba(28,37,65,0.8)',
  glassBackgroundDark: 'rgba(21,30,56,0.6)',
  glassBackgroundDarkLight: 'rgba(21,30,56,0.8)',
  
  // White overlays
  whiteOverlay: 'rgba(255,255,255,0.05)',
  whiteOverlayLight: 'rgba(255,255,255,0.1)',
  whiteOverlayMedium: 'rgba(255,255,255,0.2)',

  // iOS System Colors (using PlatformColor for automatic adaptation)
  systemBlue: Platform.OS === 'ios' ? PlatformColor('systemBlue') : '#007AFF',
  systemGray: Platform.OS === 'ios' ? PlatformColor('systemGray') : '#8E8E93',
  systemGray2: Platform.OS === 'ios' ? PlatformColor('systemGray2') : '#AEAEB2',
  systemGray3: Platform.OS === 'ios' ? PlatformColor('systemGray3') : '#C7C7CC',
  systemGray4: Platform.OS === 'ios' ? PlatformColor('systemGray4') : '#D1D1D6',
  systemGray5: Platform.OS === 'ios' ? PlatformColor('systemGray5') : '#E5E5EA',
  systemGray6: Platform.OS === 'ios' ? PlatformColor('systemGray6') : '#F2F2F7',
  label: Platform.OS === 'ios' ? PlatformColor('label') : '#000000',
  secondaryLabel: Platform.OS === 'ios' ? PlatformColor('secondaryLabel') : '#3C3C43',
  tertiaryLabel: Platform.OS === 'ios' ? PlatformColor('tertiaryLabel') : '#3C3C4399',
  quaternaryLabel: Platform.OS === 'ios' ? PlatformColor('quaternaryLabel') : '#3C3C434D',
  separator: Platform.OS === 'ios' ? PlatformColor('separator') : '#3C3C434A',
  opaqueSeparator: Platform.OS === 'ios' ? PlatformColor('opaqueSeparator') : '#C6C6C8',
  systemBackground: Platform.OS === 'ios' ? PlatformColor('systemBackground') : '#FFFFFF',
  secondarySystemBackground: Platform.OS === 'ios' ? PlatformColor('secondarySystemBackground') : '#F2F2F7',
  tertiarySystemBackground: Platform.OS === 'ios' ? PlatformColor('tertiarySystemBackground') : '#FFFFFF',
};

<<<<<<< HEAD:mobile/constants/theme.ts
// =======================
// TYPOGRAPHY
// =======================
import type { TextStyle } from 'react-native';
import { Platform } from 'react-native';

=======
>>>>>>> main:mobile/constants/theme.js
export const typography = {
  fontWeight: {
    regular: '400',
    medium: '500',
    semiBold: '600',
    bold: '700',
  },
  fontSize: {
    xs: 12,
    sm: 13,
    base: 14,
    md: 15,
    lg: 16,
    xl: 17,
    '2xl': 20,
    '3xl': 32,
  },
  lineHeight: {
    tight: 18,
    normal: 19.5,
    relaxed: 21,
    loose: 22,
    xl: 24,
    '2xl': 26,
    '3xl': 38.4,
  },
<<<<<<< HEAD:mobile/constants/theme.ts

  // REQUIRED FIXES
  heading: 32,
  body: 14,

  // iOS System Font
  fontFamily: Platform.select({
    ios: 'System',
    android: 'Roboto',
    default: 'System',
  }),
=======
>>>>>>> main:mobile/constants/theme.js
};

export const spacing = {
  xs: 4,
  sm: 6,
  md: 8,
  lg: 10,
  xl: 12,
  '2xl': 16,
  '3xl': 20,
  '4xl': 24,
  '5xl': 40,
  '6xl': 48,
};

export const borderRadius = {
  sm: 6.8,
  md: 16,
  lg: 24,
  xl: 48,
  full: 1000, // Pill shape
};

export const shadows = {
  // Primary button shadow
  primary: {
    shadowColor: '#6c63ff',
    shadowOffset: { width: 0, height: 10 },
    shadowOpacity: 0.3,
    shadowRadius: 15,
    elevation: 10,
  },
  // Card shadow
  card: {
    shadowColor: '#000',
    shadowOffset: { width: 0, height: 4 },
    shadowOpacity: 0.25,
    shadowRadius: 12,
    elevation: 8,
  },
  // Message bubble shadow
  message: {
    shadowColor: '#6c63ff',
    shadowOffset: { width: 0, height: 4 },
    shadowOpacity: 0.2,
    shadowRadius: 6,
    elevation: 4,
  },
};

export const blur = {
  light: 20,
  medium: 80,
  heavy: 100,
  xl: 120,
};

File: ios-app/VioletVibes/Resources/Theme.swift
//
//  Theme.swift
//  VioletVibes
//

import SwiftUI

struct Theme {
    // MARK: - Colors
    struct Colors {
        static let background = Color(hex: "0b132b")
        static let backgroundSecondary = Color(hex: "0d1630")
        static let backgroundCard = Color(hex: "1c2541")
        static let backgroundCardDark = Color(hex: "151e38")
        static let backgroundOverlay = Color(hex: "0b132b").opacity(0.95)
        static let backgroundPrimary = Color(hex: "0b132b")
        
        static let gradientStart = Color(hex: "6c63ff")
        static let gradientEnd = Color(hex: "8b7fff")
        static let gradientBlueStart = Color(hex: "38c4fc")
        static let gradientBlueEnd = Color(hex: "76caf5")
        
        static let textPrimary = Color(hex: "ffffff")
        static let textSecondary = Color(hex: "a9b4d1")
        static let textAccent = Color(hex: "cbb8ff")
        static let textBlue = Color(hex: "68d4ff")
        static let textError = Color(hex: "ff3b30")
        
        static let border = Color.white.opacity(0.1)
        static let borderLight = Color.white.opacity(0.05)
        static let borderMedium = Color.white.opacity(0.2)
        
        static let accentPurple = Color(hex: "6c63ff").opacity(0.1)
        static let accentPurpleMedium = Color(hex: "6c63ff").opacity(0.2)
        static let accentBlue = Color(hex: "68d4ff").opacity(0.1)
        static let accentBlueMedium = Color(hex: "68d4ff").opacity(0.2)
        static let accentPurpleText = Color(hex: "cbb8ff").opacity(0.1)
        
        static let accentError = Color(hex: "ff3b30").opacity(0.1)
        static let accentErrorMedium = Color(hex: "ff3b30").opacity(0.2)
        static let accentErrorBorder = Color(hex: "ff3b30").opacity(0.5)
        
        static let glassBackground = Color(hex: "1c2541").opacity(0.6)
        static let glassBackgroundLight = Color(hex: "1c2541").opacity(0.8)
        static let glassBackgroundDark = Color(hex: "151e38").opacity(0.6)
        static let glassBackgroundDarkLight = Color(hex: "151e38").opacity(0.8)
        
        static let whiteOverlay = Color.white.opacity(0.05)
        static let whiteOverlayLight = Color.white.opacity(0.1)
        static let whiteOverlayMedium = Color.white.opacity(0.2)
    }
    
    // MARK: - Typography
    struct Typography {
        enum FontWeight {
            case regular
            case medium
            case semiBold
            case bold
            
            var value: Font.Weight {
                switch self {
                case .regular: return .regular
                case .medium: return .medium
                case .semiBold: return .semibold
                case .bold: return .bold
                }
            }
        }
        
        enum FontSize: CGFloat {
            case xs = 12
            case sm = 13
            case base = 14
            case md = 15
            case lg = 16
            case xl = 17
            case `2xl` = 20
            case `3xl` = 32
            
            var value: CGFloat { rawValue }
        }
        
        enum LineHeight: CGFloat {
            case tight = 18
            case normal = 19.5
            case relaxed = 21
            case loose = 22
            case xl = 24
            case `2xl` = 26
            case `3xl` = 38.4
            
            var value: CGFloat { rawValue }
        }
        
        static let heading: CGFloat = 32
        static let body: CGFloat = 14
    }
    
    // MARK: - Spacing
    struct Spacing {
        static let xs: CGFloat = 4
        static let sm: CGFloat = 6
        static let md: CGFloat = 8
        static let lg: CGFloat = 10
        static let xl: CGFloat = 12
        static let `2xl`: CGFloat = 16
        static let `3xl`: CGFloat = 20
        static let `4xl`: CGFloat = 24
        static let `5xl`: CGFloat = 40
        static let `6xl`: CGFloat = 48
    }
    
    // MARK: - Border Radius
    struct BorderRadius {
        static let sm: CGFloat = 6.8
        static let md: CGFloat = 16
        static let lg: CGFloat = 24
        static let xl: CGFloat = 48
        static let full: CGFloat = 1000
    }
    
    // MARK: - Shadows
    struct Shadows {
        static let primary = Shadow(color: Color(hex: "6c63ff"), offset: CGSize(width: 0, height: 10), radius: 15, opacity: 0.3)
        static let card = Shadow(color: .black, offset: CGSize(width: 0, height: 4), radius: 12, opacity: 0.25)
        static let message = Shadow(color: Color(hex: "6c63ff"), offset: CGSize(width: 0, height: 4), radius: 6, opacity: 0.2)
    }
    
    struct Shadow {
        let color: Color
        let offset: CGSize
        let radius: CGFloat
        let opacity: Double
    }
}

// MARK: - View Modifiers
extension View {
    func themeFont(size: Theme.Typography.FontSize, weight: Theme.Typography.FontWeight = .regular) -> some View {
        self.font(.system(size: size.value, weight: weight.value))
    }
    
    func themeShadow(_ shadow: Theme.Shadow) -> some View {
        self.shadow(color: shadow.color.opacity(shadow.opacity), radius: shadow.radius, x: shadow.offset.width, y: shadow.offset.height)
    }
}

